### ISSUE TITLE
Leaking open cursors on ORA-00942

### ISSUE THREAD

**Issue by birkoffe** [sijms/go-ora#272]
After multiple selects on not exist view/table I start getting `ORA-01000: maximum open cursors exceeded` error.

```
go-ora v2.5.8
oracle 18c
```

Code to reproduce 

```
func main() {
    ...

    pool, err := sql.Open("oracle", dsn)
    defer pool.Close()

    ...

    query := "select * from NO_EXIST_TABLE"

    scheduler := gocron.NewScheduler(time.UTC)

    if _, err := scheduler.Every(100).Milisecond().Do(execQuery, pool, query); err := nil {
        log.Fatalf("Err job: %v", err)
    }

    scheduler.StartBlocking()
}

func execQuery(pool *sql.DB, query string) {
    ctx, cancel := context.WithTimeout(context.Background(), time.Duration(5)*time.Second)
    defer cancel()

    rows, err := pool.QueryContext(ctx, query)
    if err != nill {
        log.Printf("Rows err: %v", err)
    }
    defer rows.Close()

    log.Printf("%v", rows)
}
```

Logs

```
2022/11/22 10:00:00 Rows err: ORA-00942: table or view does not exist
...
2022/11/22 10:02:53 Rows err: ORA-00942: table or view does not exist
2022/11/22 10:02:53 Rows err: ORA-01000: maximum open cursors exceeded
...
2022/11/22 10:03:00 Rows err: ORA-01000: maximum open cursors exceeded
```

I try to add prepare statement in `execQuery` -- same behavior.

**Comment by sijms** [comment]
fixed in v2.5.10

**Comment by birkoffe** [comment]
Testing on v2.5.10, doesn't help.

**Comment by sijms** [comment]
can you help me if you have a go dubugger like goland
can you make a breakpoint at stmt.Close line [1348](https://github.com/sijms/go-ora/blob/772c07a992f76645a119ce16036f5c1bab990fb4/v2/command.go#L1348) function and see if the driver get cursor id or not
because it seems that Close is called but not function because no cursor id is defined

also cursor id is defined at this line [496](https://github.com/sijms/go-ora/blob/772c07a992f76645a119ce16036f5c1bab990fb4/v2/command.go#L496)

**Comment by birkoffe** [comment]
In vscode (dlv) on 1348

```
stmt
    text: "select * from NO_EXIST_TABLE"
    ...
    stmtType: SELCT (1)
    cursorID: 0
    queryID: 0 = 0x0
    ... 
```

Trying make breakpoint on 469 -- doesn't catch.

```
func (stmt *defaultStmt) read(dataSet *DataSet) error {
	loop := true
	after7 := false
	dataSet.parent = stmt
	session := stmt.connection.session
	for loop {
		msg, err := session.GetByte()
		if err != nil {
			return err       <---- return here with { ErrCode: 142, ErrMsg: "ORA-00942: table or view does not exist\n" }
		}
		switch msg {
```

**Comment by sijms** [comment]
I think it is fixed in version 2.5.12
pls test and inform me

**Comment by birkoffe** [comment]
Everything ok after a ten-minute stress testing.
Thanks!
